.github/workflows/freertos.yml

```
@@ -40,11 +40,7 @@ jobs:
   compile-check-fedora:
     name: FreeRTOS PAL compile check (Fedora 42)
     runs-on: ubuntu-latest
-    container:
-      image: fedora:42
-      credentials:
-        username: ${{ secrets.DOCKERHUB_USERNAME }}
-        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    container: ${{ secrets.DOCKERHUB_USERNAME && fromJSON(format('{{"image":"fedora:42","credentials":{{"username":"{0}","password":"{1}"}}}}', secrets.DOCKERHUB_USERNAME, secrets.DOCKERHUB_TOKEN)) || 'fedora:42' }}
```

actionlint (1.7.11)

[error] context "secrets" is not allowed here. available contexts are "github", "inputs", "matrix", "needs", "strategy", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)